packages/@lwc/synthetic-shadow/src/3rdparty/polymer/path-composer.ts

```diff
@@ -18,22 +14,29 @@ The complete set of contributors may be found at http://polymer.github.io/CONTRI
 Code distributed by Google as part of the polymer project is also
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 */
+
+import { getOwnerDocument } from '../../shared/utils';
+import { isNull } from '../../shared/language';
+
+const GlobalShadowRoot = (window as any).ShadowRoot;
+
 export function pathComposer(startNode: EventTarget, composed: boolean): EventTarget[] {
-    const composedPath: HTMLElement[] = [];
-    let current = startNode;
-    const startRoot = startNode instanceof Window ? startNode : patchedGetRootNode.call(startNode);
+    const composedPath: (Element | Document | Window)[] = [];
+    let current: Node | null = startNode as Element;
+    const startRoot: Window | Node =
+        startNode instanceof Window ? startNode : (startNode as Node).getRootNode();
     while (current) {
-        composedPath.push(current as HTMLElement);
-        if ((current as HTMLElement).assignedSlot) {
-            current = (current as HTMLElement).assignedSlot as HTMLSlotElement;
-        } else if (
-            (current as HTMLElement).nodeType === DOCUMENT_FRAGMENT_NODE &&
-            (current as ShadowRoot).host &&
-            (composed || current !== startRoot)
-        ) {
-            current = (current as ShadowRoot).host as HTMLElement;
+        composedPath.push(current as Element);
+        let assignedSlot: HTMLSlotElement | null = null;
+        if (current instanceof Element) {
+            assignedSlot = current.assignedSlot;
+        }
+        if (!isNull(assignedSlot)) {
```

Comment:
```
        if (isTrue(current instanceof Element) && !isNull((current as Element).assignedSlot!)) {
            current = (current as Element).assignedSlot;
        }
```

Reply:
`isTrue(current instanceof Element)` seems a bit too much. The way I decide whether to use `isTrue` and other variants is if I would  have otherwise written `foo === true`. For something like`current instanceof Element`, I wouldn't have written that as `(current instanceof Element) === true` so I wouldn't use `isTrue`. Thoughts?

Reply:
good q! maybe we should open an issue to track that since it is a larger change through multiple pkgs.